FileIO/OpenGeoSysProject.xsd

```diff
@@ -2,57 +2,183 @@
 <xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
 
   <!-- OpenGeoSysProject schema; initial implementation: Karsten Rink, 2010-08-20 -->
-  
+
   <!-- object containing just a source file definition -->
-  <xs:complexType name="projectElement">
+  <xs:complexType name="FileType">
     <xs:sequence>
-	<xs:element name="file" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="file" type="xs:string" minOccurs="1" maxOccurs="1" />
     </xs:sequence>
   </xs:complexType>
 
-  <!-- object containing a required reference to a geometry file -->
-  <xs:complexType name="geoElement">
+  <!-- mesh-object with optional geometry- and texture-reference -->
+  <xs:complexType name="MeshFileType">
     <xs:complexContent>
-      <xs:extension base="projectElement">
+      <xs:extension base="FileType">
         <xs:sequence>
-          <xs:element name="geoReference" type="xs:string"/>
+          <xs:element name="GeoReference" type="xs:string" minOccurs="0" maxOccurs="1"/>
+          <xs:element name="Texture" type="xs:string" minOccurs="0" maxOccurs="1" />
         </xs:sequence>
       </xs:extension>
     </xs:complexContent>
   </xs:complexType>
-  
-  <!-- msh-object with optional geometry- and texture-reference -->
-  <xs:complexType name="mshElement">
-    <xs:complexContent>
-      <xs:extension base="projectElement">
-        <xs:sequence>
-          <xs:element name="geoReference" type="xs:string" minOccurs="0" />
-          <xs:element name="texture" type="xs:string" minOccurs="0" />
-        </xs:sequence>
-      </xs:extension>
-    </xs:complexContent>
+
+  <!-- output-object with required type-reference -->
+  <xs:complexType name="OutputFileType">
+    <xs:all>
+      <xs:element name="file" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+    </xs:all>
+  </xs:complexType>
+
+  <xs:complexType name="ProcessType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="process_variable" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <!-- undefined content -->
+      <xs:any processContents="lax" maxOccurs="unbounded"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="ParameterType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="value" type="xs:decimal" minOccurs="1" maxOccurs="1" />
```

Comment:
"value" is depending on "type".
If type is MeshProperty, then there is no "value" field.